### ISSUE TITLE
ICE: trait takes 1 generic argument but 0 generic arguments were supplied

### ISSUE THREAD

**Issue by rucsesec** [rust-lang/rust#148121]
I tried this code.

### Code

```Rust
pub trait Super<X> {
    type X;
}

impl<T> Clone for Box<dyn Super<X = T>> {
    fn clone(&self) -> Self {
        unimplemented!();
    }
}

pub fn main() {}

```

### Meta

`rustc --version --verbose`:
```
rustc 1.92.0-nightly (4068bafed 2025-10-20)
binary: rustc
commit-hash: 4068bafedd8ba724e332a5221c06a6fa531a30d2
commit-date: 2025-10-20
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.3
```

<details><summary><strong>Program Output</strong></summary>
<p>

```
error: internal compiler error[E0107]: trait takes 1 generic argument but 0 generic arguments were supplied
 --> <source>:5:27
  |
5 | impl<T> Clone for Box<dyn Super<X = T>> {
  |                           ^^^^^ expected 1 generic argument
  |
note: trait defined here, with 1 generic parameter: `X`
 --> <source>:1:11
  |
1 | pub trait Super<X> {
  |           ^^^^^ -
note: delayed at compiler/rustc_hir_analysis/src/errors/wrong_number_of_generic_args.rs:1146:23 - disabled backtrace
 --> <source>:5:27
  |
5 | impl<T> Clone for Box<dyn Super<X = T>> {
  |                           ^^^^^
help: add missing generic argument
  |
5 | impl<T> Clone for Box<dyn Super<X, X = T>> {
  |                                 ++

error: internal compiler error: TyKind::Error constructed but no error reported
  |
  = note: delayed at compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs:649:29 - disabled backtrace

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: please attach the file at `/app/rustc-ice-2025-10-26T02_27_23-1.txt` to your bug report

note: compiler flags: -C debuginfo=2 --crate-type bin -C linker=/opt/compiler-explorer/gcc-15.2.0/bin/gcc

query stack during panic:
end of query stack
```

</p>
</details>

**Comment by rucsesec** [comment]
It seems to have the same root cause as #128249 which has already been fixed.

**Comment by cyrgani** [comment]
simpler:
```rust
pub trait Super<X> {
    type X;
}
pub trait A {}
impl A for dyn Super<X = ()> {}
```
